shells/android/java/arcs/storage/service/BindingContext.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.storage.service
+
+import androidx.annotation.VisibleForTesting
+import arcs.crdt.CrdtData
+import arcs.crdt.CrdtException
+import arcs.crdt.CrdtOperation
+import arcs.crdt.parcelables.ParcelableCrdtType
+import arcs.crdt.parcelables.toParcelable
+import arcs.storage.ActiveStore
+import arcs.storage.ProxyCallback
+import arcs.storage.ProxyMessage
+import arcs.storage.Store
+import arcs.storage.parcelables.ParcelableProxyMessage
+import arcs.storage.parcelables.toParcelable
+import arcs.storage.util.SendQueue
+import kotlin.coroutines.CoroutineContext
+import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.supervisorScope
+
+/**
+ * A [BindingContext] is used by a client of the [StorageService] to facilitate communication with a
+ * [Store] residing within the [StorageService] from elsewhere in an application.
+ */
+class BindingContext(
+    /**
+     * The [Store] this [BindingContext] provides bindings for, it may or may not be shared with
+     * other instances of [BindingContext].
+     */
+    private val store: Store<*, *, *>,
+    /**
+     * A hint used to facilitate translations between usable Crdt types and [Parcelable]s required
+     * for transmission via IBinder interfaces.
+     */
+    private val crdtType: ParcelableCrdtType,
+    /**
+     *
+     */
+    parentCoroutineContext: CoroutineContext
+) : IStorageService.Stub() {
+    @VisibleForTesting
+    val id = nextId.incrementAndGet()
```

Comment:
This is up to you, but if you wanted this to be `internal`, you could add it like so: `/* internal */`. I'm planning to do a big find and replace as soon as internal works internally.

Reply:
I'm going to just run with it as-is, in case it takes forever for the internal stuff to catch up.